src/Git.hs

```diff
@@ -483,19 +525,24 @@ tryIntegrate message candidateRef candidateSha targetBranch testBranch alwaysAdd
       -- If not (i.e. the current master is the parent of the proposed commit
       -- and the approval type is not MergeAndDeploy) then we just take that
       -- commit as-is.
-      targetBranchSha <- checkout targetBranch
-      parentSha       <- getParent sha
-      newTip <- case parentSha of
-        Nothing -> pure $ Just sha
-        parent
-          | alwaysAddMergeCommit      -> merge sha message
-          | parent == targetBranchSha -> pure $ Just sha
-        _moreThanOneCommitBehind      -> merge sha message
-
-      -- If both the rebase, and the (potential) merge went well, push it to the
-      -- testing branch so CI will build it.
-      case newTip of
-        Just tipSha -> forcePush tipSha testBranch
-        Nothing     -> pure ()
-
-      pure newTip
+      checkOrphansResult <- checkOrphanFixups sha targetBranch
+      if checkOrphansResult 
+        then pure $ Left WrongFixups
```

Comment:
Nit: the current naming makes the resulting boolean ambiguous in some ways. Getting `True` from the check could mean that we *passed* and are good to go, or that there are fixups and we are not good to go (the latter is the interpretation here).

We could rename it to `hasOrphanFixups`, to make it more apparent.